src/pushsource/_impl/backend/registry_source.py

```diff
@@ -108,10 +108,9 @@ def _push_item_from_registry_uri(self, uri, signing_key):
                         manifest_types=[mtype],
                     )
                     self._manifests[source_uri][mtype] = manifest_details
-                except (
-                    requests.exceptions.ConnectionError,
-                    requests.exceptions.HTTPError,
-                ) as e:
+                except requests.exceptions.RetryError as e:
```

Comment:
What's the reasoning behind this?  RetryError is what's raised after all retries are exhausted, it's not clear why this should be ignored.

Could you point at a specific problem (i.e. with traceback) which would be solved by this?

Reply:
Instead of HTTPError, RetryError is raise even on 404. But 404 should be tolerated here.

Reply:
It looks OK now.  In the version of the code where I posted this comment, you didn't check for 404 but instead caught all kinds of retryerror with no filtering.